### PR TITLE
chore: add `check_format.yml`

### PR DESCRIPTION
**Pull request by vil02**
I have noticed that formatting of some of the files is a bit off (cf. 7c18e42c9643db0ef8a66d9dd85c2a84f2793400). This PR formats some of the files and adds a workflow checking the formatting it the future.

### CODE REVIEW

**Review by uncomfyhalomacro — DISMISSED** [review]
i forgot about this but thanks!

**Comment by vil02** [comment]
Resolving conflicts, when formatting changes, can be painful. @uncomfyhalomacro could you please merge this PR?

**Comment by uncomfyhalomacro** [comment]
> Resolving conflicts, when formatting changes, can be painful. @uncomfyhalomacro could you please merge this PR?

i dont have merge rights in some circumstances in this repository tbh. @Panquesito7 has the power here to do so

**Review by uncomfyhalomacro — DISMISSED** [review]

File: .github/workflows/check_format.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  pull_request:
+
+jobs:
+  check_format:
+    name: Check format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: 1.8
```
Comment (Panquesito7) [inline .github/workflows/check_format.yml:19]:
```suggestion
          version: 1.9
```

File: .github/workflows/check_format.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          git clean -f -x -d
+          julia  -e 'using JuliaFormatter; format(".", verbose=true)'
+
+      - name: Fail if needs reformatting
+        run: |
+          if [[ $(git status --porcelain) ]]; then
+              echo "please reformat these files:"
+              git status --porcelain=v1
+              exit 1
+          fi
```
Comment (Panquesito7) [inline .github/workflows/check_format.yml:39]:
```suggestion
          fi
...
```

**Review by Panquesito7 — COMMENTED** [review]
Other than that, it looks good. Thanks! 🚀

**Review by Panquesito7 — APPROVED** [review]
Great. Thanks! 🚀

**Pull request merged**